enumerable_method.rb

```diff
@@ -0,0 +1,124 @@
+public def my_each(arr)
+    for name in arr do
+        yield (name)
+    end
+end
+
+names = ["Amy", "John", "Joe"]
+my_each(names) {|a| puts "Hello #{a}"}
+
+
+names = %w[Amy John Joe]
+
+my_each(names) { |a| puts "Hello #{a}" }
+
+def my_each_with_index(arr)
+  len = arr.length
+
+  i = 1
+
+  my_each(arr) do |item|
+    yield item if i <= len && i.even?
+
+    i += 1
+  end
+end
+
+grocery_list = %w[banana plum chocolate toiletries]
+
+my_each_with_index(grocery_list) { |m| puts "Please get #{m}" }
+
+def my_select(arr)
+  new_array = []
+    for val in arr do  
+      unless val.is_a? Integer
+        new_array.push(val)
+      end
+    end
+    p new_array
+end
+    new_list = ["a", "b", "c", 3]
+  my_select(new_list)
+  
+
+def my_all(arr)
+    a=0
+    len = arr.length
+    for f in arr do 
+        if f.is_a? Integer
+            a+=1
+        end
+    end   
+    if a == len
+        p true
+    else
+        p false
+    end
+end
+    new_element = [1, 3, 2]
+my_all(new_element)
+
+def my_any(arr)
+    for f in arr do 
+        if f.is_a? Integer
+            puts true
+        end
+    end  
+    puts false 
+end
+    new_element = ["a", "b"]
+my_all(new_element)
+
+def my_none(arr)
+    output = true
+    my_each(arr) do |num|
+        if num.is_a? Integer
+            output = false
+        end
+    end  
+    if output == false
+        p false
+    else
+        p true
+    end
+end
```

Comment:
#my_none? returns true if the block never returns true for all elements
#my_none? returns false if the block ever returns true for all elements
#my_none? returns false if the block ever returns true for all elements::range
#my_none? does not mutate the original array
#my_none? when no block or argument is given returns true only if none of the collection members is true
#my_none? when no block or argument is given returns false only if one of the collection members is true
#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
#my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
#my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
#my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
#my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
#my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern